users/vieting/experiments/switchboard/ctc/feat/network_helpers/perturbation.py

```diff
@@ -121,17 +123,21 @@ def preemphasis_numpy(waveform, coeff=0.97):
 
     @staticmethod
     def apply_codecs(audio, sample_rate, random_state, codecs):
-        import sox
-
-        tfm = sox.Transformer()
         for codec in codecs:
             prob = codec.pop("prob", 1.0)
             if random_state.random() < prob:
                 if codec.get("encoding") == "ULAW":
-                    tfm.set_output_format(encoding="u-law")
+                    # standard value for mu-law encoding
+                    mu = 255.0
+
+                    # mu-law encoding formula
+                    encoded_audio = np.sign(audio) * np.log1p(mu * np.abs(audio)) / np.log1p(mu)
+                    encoded_audio = ((encoded_audio + 1) / 2 * 255).astype(np.float32)
+
+                    audio = encoded_audio
```

Comment:
Is the `encoded_audio` variable really necessary?

Reply:
No, but it makes the code more readable especially if more encodings are added in the future.

Reply:
Do you still think it improves the readability? :thinking:

Reply:
I removed it for now. As long as there is no other encoding it should be clear.